src/osd/osd_types.cc

```diff
@@ -1311,7 +1311,8 @@ bool pg_pool_t::maybe_updated_removed_snaps(const interval_set<snapid_t>& cached
 }
 
 void pg_pool_t::purged_snap(const snapid_t& purged) {
-  deleting_snaps.erase(purged);
+  if (deleting_snaps.contains(purged, 1))
+    deleting_snaps.erase(purged);
```

Comment:
It may be worth doing the iterator-based find() and erase, instead of having to look it up twice.
Normally I wouldn't fuss about something like that, but since you're having CPU issues dealing with the size of these sets... ;)

Reply:
Sounds good. However, in order to do this cleanly, it looks like we'll have to change the `interval_set::erase` signature to `void erase(T start, T len, bool ignore_missing=false)` or something like that.

Reply:
Oh boo, I was just thinking in terms of the STL. Presumably we'd want to add an iterator-based erease() rather than remove the existing interface though.